scripts/macros/spells/spiritOfDeath.js

```diff
@@ -0,0 +1,221 @@
+        if (concentrationEffect) await genericUtils.remove(concentrationEffect);
+        return;
+    }
+    let name = itemUtils.getConfig(workflow.item, 'name');
+    if (!name?.length) name = genericUtils.translate('CHRISPREMADES.Summons.CreatureNames.ReaperSpirit');
+    let updates = {
+        actor: {
+            name,
+            system: {
+                details: {
+                    cr: actorUtils.getCRFromProf(workflow.actor.system.attributes.prof)
+                },
+                attributes: {
+                    ac: {
+                        flat: 11 + spellLevel
+                    },
+                }
+            },
+            prototypeToken: {
+                name,
+                disposition: workflow.token.document.disposition
+            },
+            items: [multiAttackFeatureData, reapingScytheFeatureData, hauntCreatureFeatureData]
+        },
+        token: {
+            name,
+            disposition: workflow.token.document.disposition
+        }
+    };
+    let avatarImg = itemUtils.getConfig(workflow.item, 'avatar');
+    let tokenImg = itemUtils.getConfig(workflow.item, 'token');
+    if (avatarImg) updates.actor.img = avatarImg;
+    if (tokenImg) {
+        genericUtils.setProperty(updates, 'actor.prototypeToken.texture.src', tokenImg);
+        genericUtils.setProperty(updates, 'token.texture.src', tokenImg);
+    }
+    let hpFormula = 40 + (spellLevel - 4) * 10;
+    updates.actor.system.attributes.hp = {
+        formula: hpFormula,
+        max: hpFormula,
+        value: hpFormula
+    };
+    let animation = itemUtils.getConfig(workflow.item, 'animation') ?? 'none';
+    await Summons.spawn(sourceActor, updates, workflow.item, workflow.token, {
+        duration: 3600,
+        range: 90,
+        animation,
+        initiativeType: 'follows',
+        additionalSummonVaeButtons: [multiAttackFeatureData, reapingScytheFeatureData, hauntCreatureFeatureData].map(i => {return {type: 'use', name: i.name, identifier: i.flags['chris-premades'].info.identifier};})
+    });
+}
+async function attack({workflow}) {
+    if (workflow.targets.size !== 1 || workflow.advantage) return;
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'spiritOfDeathHauntCreature');
+    if (!effect) {
+        workflow.aborted = true;   
+        return;
+    }
+    let {targets: validTargetUuids, formula} = effect.flags['chris-premades'].spiritOfDeathHauntCreature;
+    if (!validTargetUuids.includes(workflow.targets.first().document.uuid)) {
+        workflow.aborted = true;
+        return;
+    }
+    workflow.advantage = true;
+    workflow.attackAdvAttribution.add(genericUtils.translate('DND5E.Advantage') + ': ' + effect.name);
+}
+async function late({workflow}) {
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'summonedEffect');
+    let originItem = fromUuidSync(effect?.origin);
+    let sourceActor = originItem?.actor;
+    if(!sourceActor)
+    {
```

Comment:
How dare you. (In this case actually, no need for brackets at all. `if (!sourceActor) return;`